html/browsers/windows/clear-window-name.https.html

```diff
@@ -18,91 +18,82 @@
   hyperlink.click();
 }
 
-function pollResultAndCheck(t, id, expected) {
-  return new Promise(resolve => {
-    const stashURL = new URL("resources/window-name-stash.py", location);
-    stashURL.searchParams.set('id', id);
-
-    let checkResult = t.step_func(async () => {
-      let response = await fetch(stashURL);
-      let res = await response.text();
-
-
-      if (res !== "NONE") {
-        if (res !== expected) {
-          t.step(() => assert_unreached('Stash result does not equal expected result.'));
-        }
-        t.done();
-        resolve();
-      } else {
-        t.step_timeout(checkResult, 100);
-      }
-    });
-
-    t.step_timeout(checkResult, 100);
-  });
+async function pollResultAndCheck(t, id, expected) {
+  const stashURL = new URL("resources/window-name-stash.py", location);
+  stashURL.searchParams.set('id', id);
+
+  let res = "NONE";
+  while (res == "NONE") {
+    await new Promise(resolve => { t.step_timeout(resolve, 100); });
+
+    const response = await fetch(stashURL);
+    res = await response.text();
+  }
+  if (res !== expected) {
```

Comment:
Just `assert_equals(res, expected, "bla bla")` would result in a better message on failure.

Reply:
This was deliberately changed from that in 5a5410b369b69e783fbb526eb997cd70c90ded8e, to avoid producing a different output every time and thus upsetting the Chromium baseline comparison, so leaving as-is.